Sources/GRPCProtobufParser/ProtobufToCodeGenParser.swift

```diff
@@ -0,0 +1,122 @@
+    let lookupSerializer: (String) -> String = { messageType in
+      "ProtobufSerializer<\(messageType)>()"
+    }
+    let lookupDeserializer: (String) -> String = { messageType in
+      "ProtobufDeserializer<\(messageType)>()"
+    }
+    let services = input.services.map {
+      CodeGenerationRequest.ServiceDescriptor(descriptor: $0, package: input.package)
+    }
+
+    return CodeGenerationRequest(
+      fileName: input.name,
+      leadingTrivia: leadingTrivia + header,
+      dependencies: dependencies,
+      services: services,
+      lookupSerializer: lookupSerializer,
+      lookupDeserializer: lookupDeserializer
+    )
+  }
+}
+
+extension CodeGenerationRequest.ServiceDescriptor {
+  fileprivate init(descriptor: ServiceDescriptor, package: String) {
+    let methods = descriptor.methods.map {
+      CodeGenerationRequest.ServiceDescriptor.MethodDescriptor(descriptor: $0)
+    }
+    let name = CodeGenerationRequest.Name(
+      base: descriptor.name,
+      generatedUpperCase: NamingUtils.toUpperCamelCase(descriptor.name),
+      generatedLowerCase: NamingUtils.toLowerCamelCase(descriptor.name)
+    )
+    let namespace = CodeGenerationRequest.Name(
+      base: package,
+      generatedUpperCase: NamingUtils.toUpperCamelCase(package),
+      generatedLowerCase: NamingUtils.toLowerCamelCase(package)
+    )
+    let documentation = descriptor.protoSourceComments()
+    self.init(documentation: documentation, name: name, namespace: namespace, methods: methods)
+  }
+}
+
+extension CodeGenerationRequest.ServiceDescriptor.MethodDescriptor {
+  fileprivate init(descriptor: MethodDescriptor) {
+    let name = CodeGenerationRequest.Name(
+      base: descriptor.name,
+      generatedUpperCase: NamingUtils.toUpperCamelCase(descriptor.name),
+      generatedLowerCase: NamingUtils.toLowerCamelCase(descriptor.name)
+    )
+    let documentation = descriptor.protoSourceComments()
+    self.init(
+      documentation: documentation,
+      name: name,
+      isInputStreaming: descriptor.clientStreaming,
+      isOutputStreaming: descriptor.serverStreaming,
+      inputType: descriptor.inputType.name,
+      outputType: descriptor.outputType.name
+    )
+  }
+}
+
+extension FileDescriptor {
+  fileprivate func getHeader() -> String {
+    var header = String()
+    // Field number used to collect the syntax field which is usually the first
+    // declaration in a.proto file.
+    // See more here:
+    // https://github.com/apple/swift-protobuf/blob/main/Protos/SwiftProtobuf/google/protobuf/descriptor.proto
+    let syntaxPath = IndexPath(index: 12)
+    if let syntaxLocation = self.sourceCodeInfoLocation(path: syntaxPath) {
+      header = syntaxLocation.asSourceComment(
+        commentPrefix: "///",
+        leadingDetachedPrefix: "//"
+      )
+      // Ensuring there is a blank line after the header.
+      if !header.isEmpty && !header.hasSuffix("\n\n") {
+        header.append("\n")
+      }
```

Comment:
This logic is separate from getting the header, looks like it should be in `parse`